### PR TITLE
Document Versioning Schemes

### PR DESCRIPTION
**Pull request by bourque**
# Change Summary

## Overview
This PR updates the `Versioning` page of the style guide to reflect the software (#43) and data product (#48) versioning schemes that we decided on.

## Updated Files

- `versioning.rst`
   - Added details of the versioning schemes we want to use

### CODE REVIEW

File: docs/source/development/style_guide/versioning.rst [hunk 1]
```diff
@@ -3,11 +3,38 @@
...
-    z = The hotfix number. Increase when change only contains bug fixes.
\ No newline at end of file
+    XX = The major version. Increase when any change to the processing algorithm is made.
+    YY = The minor version. Increase when data are reprocessed with the same algorithm, but with different ancillary
+         files or calibrations (for example, updated SPICE kernels). The minor version is also reset to 0 whenever the
+         major version number is updated.
+
+Version numbers are specific to an instrument. In other words, each IMAP instrument has their own version number that
+is only updated when that instrument's algorithm is changed, or that instrument's data are reprocessed. The version is
+updated independently of the other IMA1P instruments.
```
Comment (tech3371) [inline docs/source/development/style_guide/versioning.rst]: Looks like there is a Typo. Did you mean `IMAP`?

**Review by tech3371 — APPROVED** [review]
Besides that, it looks good to me!

File: docs/source/development/style_guide/versioning.rst (hunk 1 above)
Comment (bourque) [reply docs/source/development/style_guide/versioning.rst]: Good catch, thanks!

**Review by sdhoyt — APPROVED** [review]
Looks great!

**Pull request merged**